core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala

```diff
@@ -0,0 +1,170 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.spark.h2o.backends.external
+
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.h2o.{H2OConf, H2OContext}
+import org.apache.spark.h2o.backends.SparklingBackend
+import org.apache.spark.h2o.utils.NodeDesc
+import org.apache.spark.internal.Logging
+import water.api.RestAPIManager
+import water.{H2O, H2OStarter}
+
+import scala.util.Random
+import scala.util.control.NoStackTrace
+
+
+class ExternalH2OBackend(val hc: H2OContext) extends SparklingBackend with ExternalBackendUtils with Logging {
+
+  private var yarnAppId: String = _
+  def launchH2OOnYarn(conf: H2OConf): (String, String) = {
+    val cmdToLaunch = Seq[String](
+      "hadoop",
+      "jar",
+      conf.h2oDriverPath.get,
+      conf.YARNQueue.map("-Dmapreduce.job.queuename=" + _ ).getOrElse(""),
+      "-nodes", conf.numOfExternalH2ONodes.get,
+      "-J", "-md5skip",
+      "-mapperXmx", conf.mapperXmx,
+      "-output", conf.HDFSOutputDir.get,
+      "-disown" // close the driver after cluster formation
+    )
+
+    logDebug("Command used to start H2O on yarn: " + cmdToLaunch.mkString)
+
+    import scala.sys.process._
+    val stdout = new StringBuffer()
+    val stderr = new StringBuffer()
+
+    val proc = cmdToLaunch.mkString(" ").!(ProcessLogger(
+      { msg =>
+        stdout.append(msg + "\n")
+        println(msg)
+      }, {
+        errMsg =>
+          stderr.append(errMsg + "\n")
+      }))
+
+    logDebug(stdout.toString)
+    logDebug(stderr.toString)
+
+    // parse required information
+    val clusterName = stdout.toString.split("\n").find(_.contains("Job name")).get.split("'")(1)
+    val arbitraryNodeIPPort = stdout.toString.split("\n").find(_.contains("H2O node")).get.split(" ")(2)
+    val split = stdout.toString.split("\n").find(_.contains("For YARN users, logs command is")).get.split(Array(' ','''))
+    yarnAppId = split(split.length-1)
+    assert(proc == 0, s"Process ended with return value $proc")
```

Comment:
If you say what exactly process ended badly, users will be less puzzled by this message.